app/services/ReservationService.java

```diff
@@ -0,0 +1,151 @@
+package services;
+
+import models.Reservation;
+import models.Restaurant;
+import models.RestaurantTable;
+import play.Logger;
+import repository.ReservationRepository;
+
+import javax.inject.Inject;
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by lapa on 5/25/17.
+ */
+public class ReservationService implements BaseService {
+
+    private final Long HALF_HOUR = 1800000L;
+    private final Long HOUR = 3600000L;
+    private final Long RESERVATION_PERIOD = 7200000L;
+    private Long FIRST_RESERVATION = null;
+    private Long LAST_RESERVATION = null;
+
+
+    private ReservationRepository reservationRepository;
+
+    @Inject
+    public void setReservationRepository(ReservationRepository reservationRepository) {
+        this.reservationRepository = reservationRepository;
+    }
+
+
+    public Restaurant getRestaurantById(Long id) {
```

Comment:
Inject `RestaurantService` here and use `getById` method instead of implementing same method twice.